packages/extension-base/src/background/handlers/Extension.ts

```diff
@@ -408,13 +412,16 @@ export default class Extension {
         .createType('ExtrinsicPayload', payload, { version: payload.version })
         .sign(pair);
 
-    if (savePass) {
+    if (savePass && password) {
       // unlike queued.account.address the following
       // address is encoded with the default prefix
       // which what is used for password caching mapping
-      this.#cachedUnlocks[pair.address] = Date.now() + PASSWORD_EXPIRY_MS;
+
+      await chromeStorage.setItem('password-cache', (savedPasswords) => (
+        { ...savedPasswords, [pair.address]: { password, expiresAt: Date.now() + PASSWORD_EXPIRY_MS } }
+      ));
     } else {
-      pair.lock();
+      await this.refreshAccountPasswordCache(pair, { clearCache: true });
```

Comment:
okay, it was my idea, sorry, but I think there is too much going on in `refreshAccountPasswordCache` :sweat_smile: 

How about a `this.clearCache(pair)` method and no `clearCache` param from `refreshAccountPasswordCache`? I just find it confusing that we refresh account twice

Reply:
Sure, sounds good

Reply:
Done